tests/test_asciidoc.py

```diff
@@ -0,0 +1,604 @@
+# This file is part of python-markups test suite
+# License: 3-clause BSD, see LICENSE file
+# Copyright: (C) Dmitry Shachnev, 2012-2021
+
+import unittest
+from markups.asciidoc import AsciidocMarkup
+
+
+@unittest.skipUnless(
+    AsciidocMarkup.available(),
+    'asciidoc3.py and/or lxml not available')
+class AsciidocTextTest(unittest.TestCase):
+    def test_basic(self):
+        self.maxDiff = None
+        markup = AsciidocMarkup()
+        converted = markup.convert(Basic_text)
+        text = converted.get_document_body()
+        title = converted.get_document_title()
+        stylesheet = converted.get_stylesheet()
+        stylesheet = stylesheet.replace('\r\n', '\n')
+        html_expected = Basic_html
+        title_expected = 'Hello, world!'
+        stylesheet_expected = Basic_stylesheet
+        # debug
+        # with open('junk.html', 'w') as outfile:
+        #     outfile.write(text)
+        # import pdb; pdb.set_trace()
+        # end debug
+        # using lxml requires replacing EOL chars
+        text = text.replace('\r\n', '\n')
+        self.assertEqual(html_expected[:500], text[:500])
+        self.assertEqual(title_expected, title)
+        self.assertEqual(stylesheet_expected[:300], stylesheet[:300])
+
+
+#
+# =================================================================
+#
+# Data to be used for comparison of correct results.
+#
+
+Basic_text = '''\
```

Comment:
Please use either all-lowercase or all-uppercase names for constants, not mixed case.